Sources/GRPC/AsyncAwaitSupport/GRPCAsyncStream.swift

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if compiler(>=5.5)
+
+/// This is currently a wrapper around AsyncThrowingStream because we want to be
+/// able to swap out the implementation for something else in the future.
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public struct GRPCAsyncStream<Element>: AsyncSequence {
```

Comment:
@glbrntt in the client PR you asked for us to rename this stream to be client specific, i.e. `GRPCAsyncResponseStream`.

Would you like this PR to do the same, i.e. this would be a `GRPCAsyncRequestStream` that is provided to the user handler?

Reply:
Yes please, `GRPCAsyncRequestStream` sounds good. 

(I'm wondering whether we should drop the "Async" from a bunch of places, `async`/`await` are implementation details which aren't necessarily relevant. Either way I think we should defer that discussion for later.)

Reply:
I speculatively made this change anyway: 05951c05.